### PR TITLE
mobile: Fix the mobile_release CI workflow

### PR DESCRIPTION
**Pull request by abeyad**
The `uses` field can't be defined along with the `steps` field, see https://github.com/envoyproxy/envoy/actions/runs/5687706494/workflow. This commit moves the `uses` field to the `env` section, bringing the config in line with the other mobile-*.yml configs.

### CODE REVIEW

**Comment by phlax** [comment]
> The uses field can't be defined along with the steps field

in this case the _env.yml workflow creates separate job/s (its a reusable workflow rather than an action)

File: .github/workflows/mobile-release.yml [hunk 1]
```diff
@@ -15,7 +20,6 @@ jobs:
               || !contains(github.actor, '[bot]'))
       }}
     name: android_release_artifacts
-    uses: ./.github/workflows/_env.yml
     runs-on: ubuntu-22.04
```
Comment (phlax) [inline .github/workflows/mobile-release.yml:23]:
```suggestion
    runs-on: ubuntu-22.04
    needs:
    - env
```

for the  output of the `env` job to be used (ie `needs.env.outputs...`) in a subsequent job, it needs `needs`

Comment (abeyad) [reply .github/workflows/mobile-release.yml:24]: thanks @phlax !  done!

**Review by phlax — APPROVED** [review]
lgtm, thanks @abeyad

**Comment by abeyad** [comment]
Not sure why I keep getting test coverage failures on the CI job: https://dev.azure.com/cncf/envoy/_build/results?buildId=144518&view=logs&j=5035ce29-be3c-5fcf-dc50-abb6a265f8b1&t=fa1604b8-8cc1-5fec-21c7-5a4fad8d70ea&l=4954.  Will try merging in main to see if it helps.

**Pull request merged**